### DISCUSSION TITLE
copy & paste behavior

### DISCUSSION THREAD

**Discussion by chiffreKing** [NationalSecurityAgency/ghidra#3312] [General]
Hi guys

I'm not quite sure if this belongs here but I found it kind of interesting and I'm sure that many people could get the same problem.
So what I did was creating a python script in the basic editor. Then I wanted to work with operands. Because I'm new to working with Ghidra I searched for the methods that I wanted to use on the official Ghidra page. In this case the one about the instruction class: https://ghidra.re/ghidra_docs/api/ghidra/program/model/listing/Instruction.html#getOperandType(int)
I found the method getOperandType(int opIndex) and I wanted to try it out so I copied pasted the method in my script and tried to print out a the type of a register. The result was an error   
   
File "<string>", line None
SyntaxError: Non-ASCII character in file 'xx\xx\xx.py', but no encoding declared; see http://www.python.org/peps/pep-0263.html for details

After hours of trying to understand and fix this error. I recognized that the error is not due to wrong coding. When I type in the line of code manually without copy/pasting the same line of code it works. I was able to reproduce this behavior. The problem occurs by not only copying this part: getOperandType . It occurs by copying the whole method: getOperandType(int opIndex) . So there must be some non-ascii characters in the methods from the ghidra page.

I know this is not a bug of Ghidra but I think it would be helpful if on the Ghidra page are only ascii characters. At least in the methods people want to work with. I think it is pretty common to just copy and paste the method and later modify it

**Comment by dev747368** [comment]
First, the ghidra.re website is not run by our team, we don't control the content there.

I was able to duplicate what you described, and it comes down to copy / pasting from a html source, that contains 'hidden' unicode characters such as U+200B ZERO WIDTH SPACE character, to a text document.

I haven't investigated deeper, but maybe we aren't specifying UTF8 as encoding of the source document when running the script, but I'm not sure that would completely eliminate the problems you would have with the text you got by pasting from there.

**Reply by astrelsky** [reply]
> First, the ghidra.re website is not run by our team, we don't control the content there.
> 

You should give this a shot :wink: https://github.com/NationalSecurityAgency/ghidra/settings/pages

Also this does occur with the ghidra docs from the distribution. I *think* this occurs with all javadocs but I'm not sure. It's just a zero width space between the method name and the first `(` I think.

**Reply by dev747368** [reply]
> You should give this a shot  https://github.com/NationalSecurityAgency/ghidra/settings/pages

:robot: It doesn't look like anything to me.  :robot:

**Reply by astrelsky** [reply]
> > You should give this a shot  https://github.com/NationalSecurityAgency/ghidra/settings/pages
> 
> :robot: It doesn't look like anything to me.  :robot: 
> 

I forged the link off of the one from one of my repos. There should be something to setup github pages in the settings that you can use to host the documentation. You don't even need to create a new repo or anything it can be done in a separate branch.

**Reply by dev747368** [reply]
:sigh:  Twas just a jest.  A meme'ish reference to a quote from westworld.

I agree it would be nice to use that.  Maybe now that we're hosting the binary releases here on github something like this could happen.

**Reply by astrelsky** [reply]
> :sigh: Twas just a jest. A meme'ish reference to a quote from westworld.
> 
> I agree it would be nice to use that. Maybe now that we're hosting the binary releases here on github something like this could happen.

I wasn't sure, not a movie guy. Sarcasm is hard on the internet lol.

**Comment by MPINTERNATIONAL** [comment]
Bolo for spies! How easy to do with this... fyi
